aspnetcore-webapi/GettingStartedASPNETMongoDB/GettingStartedASPNETMongoDB/Controllers/StudentsController.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using GettingStartedASPNETMongoDB.Interfaces;
+using GettingStartedASPNETMongoDB.Models;
+
+namespace GettingStartedASPNETMongoDB.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class StudentsController : ControllerBase
+    {
+        private readonly IStudentService _studentService;
+
+        private readonly ICourseService _courseService;
+
+        public StudentsController(IStudentService studentService, ICourseService courseService)
+        {
+            _studentService = studentService;
+
+            _courseService = courseService;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Student>>> GetAll()
+        => Ok(await _studentService.GetAll());
+
+        [HttpGet("{id:length(24)}")]
+        public async Task<ActionResult<Student>> GetById(string id)
+        {
+            var student = await _studentService.GetById(id);
+
+            if (student == null)
+            {
+                return NotFound();
+            }
+
+            if (student.Courses?.Count > 0)
+            {
+                var courseList = new List<Course>();
+
+                foreach (var course in student.Courses)
+                {
+                    var studentCourse = await _courseService.GetById(course);
+
+                    if (studentCourse != null)
+                    {
+                        courseList.Add(studentCourse);
+                    }
+                }
+
+                student.CourseList = courseList;
+            }
+
+            return Ok(student);
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Create(Student student)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest("Invalid request object");
+            }
+
+            await _studentService.Create(student);
+
+            return CreatedAtAction(nameof(GetById),
+                new { id = student.Id },
+                student);
```

Comment:
We should be using the `Student` returned by the `Create` call to populate the ID and student object data that are being included in the result.